src/protocol/sort/bit_permutations.rs

```diff
@@ -25,17 +25,18 @@ impl<'a, F: Field> BitPermutations<'a, F> {
     /// 2. calculate cumulative sum at each vector row
     /// 3. return back tuple of step 1 and step 2 output
     #[allow(clippy::cast_possible_truncation)]
-    fn prepare_mult_inputs<M: Mesh, G: Gateway<M, IPAProtocolStep>>(
+    fn prepare_mult_inputs<FABRIC: Fabric<IPAProtocolStep>>(
         &self,
-        ctx: &ProtocolContext<'a, G, IPAProtocolStep>,
+        ctx: &ProtocolContext<'a, IPAProtocolStep, FABRIC>,
```

Comment:
I think that we probably need to decide on some conventions for the different generic attributes.  I see that you have F for Fabric in some places, but that conflicts with Field.

I don't have any new ideas, but I wanted to flag it.

Reply:
yea I hate that too. for ambiguous generic parameters, should we always use long names? i.e. for field and fabric we'll aways say `FIELD`/`FABRIC` and `F` will be discouraged to use

Reply:
Can we look for different names?  s/Fabric/Network (so we can use N) perhaps?

Reply:
it is done now